### DISCUSSION TITLE
[help] About seed

### DISCUSSION THREAD

**Discussion by seongwon17** [ropensci/targets#1017] [Help] [answered]
### Help

- [X] I understand and agree to <https://books.ropensci.org/targets/help.html>.

### Description

Hello! 

I have two questions about seed. 
Why did you set each target to have a different seed instead of having same seed? What is the advantage for that?
How can I set the seed globally so that every target has same seed?

**Comment by wlandau** [comment, marked as answer]
Some seed customization is possible through the `seed` argument of `tar_option_set()`: https://docs.ropensci.org/targets/reference/tar_option_set.html#arguments 

In a non-`targets` imperative R script, we would expect the random number generator state to change every time `rnorm()` etc. is called. But in a pipeline, different targets could [run in different R sessions](https://books.ropensci.org/targets/hpc.html), and they could run in a non-deterministic order not know in advance. (The actual execution order depends on the dependency graph and the relative runtimes of the targets.) In spite of these challenges, the seed should still vary in a consistent way as the pipeline progresses. That is why by default each target has its own deterministic random seed that depends on the target name.

Some pipelines perform simulation studies where the same command is run in hundreds of different targets. The purpose there is to emulate random variability, and that is not possible when each simulation replication has the same seed and thus produces the same answer.

**Reply by seongwon17** [reply]
Thanks! I got it.
